erts/emulator/beam/erl_db_util.c

```diff
@@ -2374,11 +2375,8 @@ Eterm db_prog_match(Process *c_p,
 	    esp[-1] = t;
 	    break;
 	case matchCall3:
-	    bif = (Eterm (*)(Process*, ...)) *pc++;
-	    bif_args[0] = esp[-1];
-	    bif_args[1] = esp[-2];
-	    bif_args[2] = esp[-3];
-	    t = (*bif)(build_proc, bif_args);
+	    bif_call_3 = (BIF_RETTYPE (*)(BIF_ALIST_3)) *pc++;
+	    t = (*bif_call_3)(build_proc, esp-1, bif_args);
```

Comment:
No, you need to preserve the argument copy into bif_args, and pass bif_args as the second parameter.

Reply:
Yes, because the elements in esp[] are in the wrong order. The same goes for `matchCall2`. The third argument should be NULL.